salesforce/http.go

```diff
@@ -0,0 +1,23 @@
+package salesforce
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/amp-labs/connectors/common"
+	"golang.org/x/oauth2"
+)
+
+func newHTTPClient(ctx context.Context, params *sfParams) common.HTTPClient { //nolint:ireturn
+	if params.client == nil {
+		params.client = http.DefaultClient
+	}
+
+	ctx = context.WithValue(ctx, oauth2.HTTPClient, params.client)
+
+	if params.tokenSource != nil {
+		return oauth2.NewClient(ctx, params.tokenSource)
```

Comment:
Can you add a comment that the NewClient generated by oauth2 automatically refreshes tokens when necessary?

Reply:
Yeah can do